Consider Linux packages / binaries.

Consider creating binary packages for Linux.  We can host them in GCS or bintray.  We can have CMake create them, but it might be better to manually create them to spell out the build dependencies and install dependencies.

At least the following should be considered, and if rejected say why:

- [ ] Docker image with all dependencies pre-installed
- [ ] tar.gz: for Ubuntu, Fedora, **and** CentOS
- [ ] Linuxbrew: https://linuxbrew.sh
- [ ] Conan: https://conan.io
- [ ] Hunter: https://github.com/ruslo/hunter
- [ ] cget: http://cget.readthedocs.io/en/latest/
- [ ] .rpm: http://rpm.org/
- [ ] .deb: https://wiki.debian.org/Packaging/Intro?action=show&redirect=IntroDebianPackaging
- [ ] Spack: https://www.spack.io/
- [ ] Buckaroo: https://buckaroo.pm/

The RPM's are going to be put on hold for now. It fails due to this:

```
CMake Error at cmake/IncludeProtobuf.cmake:51 (find_package):
  By not providing "Findprotobuf.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "protobuf",
  but CMake did not find one.

  Could not find a package configuration file provided by "protobuf" with any
  of the following names:

    protobufConfig.cmake
    protobuf-config.cmake

  Add the installation prefix of "protobuf" to CMAKE_PREFIX_PATH or set
  "protobuf_DIR" to a directory containing one of the above files.  If
  "protobuf" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  cmake/IncludeGrpc.cmake:53 (include)
  CMakeLists.txt:151 (include)
```

When using `-DGOOGLE_CLOUD_CPP_PROTOBUF_PROVIDER=package` it fails at RPM build time.

--- a/doc/adr/2019-03-26-binary-distributions.md
+++ b/doc/adr/2019-03-26-binary-distributions.md
@@ -14,7 +14,8 @@ rpm, vcpkg).
 will not test binary distributions of our code. And we will not host configs
 for binary distributions, since that would involve hosting files which we do
 not test. However, we will do our best to make our code easily packageable by
-others in a wide variety of formats. For more context, see #333.
+others in a wide variety of formats. For more context, see
+[#333](https://github.com/googleapis/google-cloud-cpp/issues/333).
 
 **Consequences**: This decision will shield from endorsing any particular
 binary package management system. Other individuals who want to build and

--- a/doc/adr/2019-03-26-binary-distributions.md
+++ b/doc/adr/2019-03-26-binary-distributions.md
@@ -1,0 +1,24 @@
+**Title**: We will not distribute/own/maintain binary packages, but we will
+support others doing so within reason.
+
+**Status**: in review
+
+**Context**: Compiling google-cloud-cpp from source is not always possible nor
+desired by all users. Some users may want to install a binary package that
+contains our public headers and pre-compiled libraries so that they can link
+against our library. There is no single standard binary package manager in C++,
+instead there are a variety of others that users might want to use (e.g., dpkg,
+rpm, vcpkg).
+
+**Decision**: We will not directly support nor provide any binary packages. We
+will not test binary distributions of our code. And we will not host configs
+for binary distributions, since that would involve hosting files which we do
+not test. However, we will do our best to make our code easily packageable by
+others in a wide variety of formats. For more context, see #333.
+
+**Consequences**: This decision will shield from endorsing any particular
+binary package management system. Other individuals who want to build and
+maintain a binary distribution of our code may do so without our permission or
+knowledge. If they need small and reasonable tweaks from us to support their
+binary distributions, they may file an issue and we'll do our best to
+accommodate them within reason.

